libcontainer/rootfs_linux.go

```diff
@@ -1089,7 +1089,21 @@ func mountPropagate(m *configs.Mount, rootfs string, mountLabel string, mountFd
 	if err := utils.WithProcfd(rootfs, m.Destination, func(procfd string) error {
 		return mount(source, m.Destination, procfd, m.Device, uintptr(flags), data)
 	}); err != nil {
-		return err
+		if m.Device == "sysfs" {
+			/* If we are running in a user namespace, just bind mount /sys if creating
+			   sysfs failed.  */
+			ret := utils.CheckRunningInUserNamespace()
+			if ret < 0 {
+				return err
+			}
+			if err = utils.WithProcfd(rootfs, m.Destination, func(procfd string) error {
+				return mount("/sys", m.Destination, procfd, "/sys", unix.MS_BIND|unix.MS_REC|unix.MS_SLAVE, data)
```

Comment:
What if flags contain MS_RDONLY

Reply:
> What if flags contain MS_RDONLY

Yes, but not necessary, because this is in a rootless environment

Reply:
/sys main contain user-writable directories especially under /sys/fs/cgroup.
So it has to be mounted as recursively read-only (`mount_setattr(2)` with `AT_RECURSIVE`)

Reply:
And needs integration tests

Reply:
ok, i try